scripts/zephyr_module.py

```diff
@@ -215,20 +220,52 @@ def main():
     cmake = ""
     sanitycheck = ""
 
+    dep_modules = []
+    sorted_modules = []
+    start_modules = []
+
     for project in projects:
         # Avoid including Zephyr base project as module.
         if project == os.environ.get('ZEPHYR_BASE'):
             continue
 
         meta = process_module(project)
         if meta:
-            kconfig += process_kconfig(project, meta)
-            cmake += process_cmake(project, meta)
-            sanitycheck += process_sanitycheck(project, meta)
+            section = meta.get('build', dict())
+            deps = section.get('depends', [])
+            if not deps:
+                start_modules.append( [project, meta] )
+            else:
+                dep_modules.append( [project, meta, deps] )
         elif project in extra_modules:
             sys.exit(f'{project}, given in ZEPHYR_EXTRA_MODULES, '
                      'is not a valid zephyr module')
 
+    # This will do a topology search to ensure the modules are ordered
+    # according to dependency settings.
+    while start_modules:
+        # We want to preserve order as close to that returned by west.
+        # Thus remove elements from beginning, instead of pop().
```

Comment:
`node = start_modules.pop(0)`

Reply:
Why have I always been under the impression I could only `pop()` or `pop(object)`, and not based on index. 🤔

Reply:
fixed